modules/collection/programs/hyprlock.nix

```
@@ -0,0 +1,50 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}: let
+  inherit (lib.modules) mkIf;
+  inherit (lib.options) mkEnableOption mkPackageOption mkOption;
+  inherit (lib.rum.generators.hypr) toHyprconf;
+  inherit (lib.rum.types) hyprType;
+
+  cfg = config.rum.programs.hyprlock;
+in {
+  options.rum.programs.hyprlock = {
+    enable = mkEnableOption "hyprlock";
+
+    package = mkPackageOption pkgs "hyprlock" {};
+
+    settings = mkOption {
+      type = hyprType;
+      default = {};
+      example = {
+        general = {
+          hide_cursor = true;
+          no_fade_in = false;
+        };
+
+        background = [
+          {
+            path = "screenshot";
+            blur_passes = 3;
+            blur_size = 8;
+          }
+        ];
+      };
+      description = ''
+        Is written to `${config.directory}/hypr/hyprlock.conf`.
```

```suggestion
        Is written to `''${config.directory}/hypr/hyprlock.conf`.
```

This is not in the contributing guidelines yet, but we discovered recently that keeping our options impure (i.e. relying on `config`) will break when trying to evaluate on non NixOS (we cannot evaluate it for documentation).

It should get in the guidelines *soon*, when #31 gets merged.

---

Fixed that

---

I should get out of the habit to amend commits...